beacon-chain/sync/initial-sync/blocks_fetcher_test.go

```diff
@@ -0,0 +1,362 @@
+package initialsync
```

Comment:
One suggestion I have is to add more granular tests if you end up splitting up a lot of the large functions in block_fetcher.go. This would give you greater control of ensuring certain things fail as expected, especially the concurrent functions. It's always nice indeed to have one big function that tests the full behavior of a service runtime, so no need to delete it, but would be nice to have smaller tests as well.